frontend/src/components/Header/SearchInput.js

```diff
@@ -1,96 +1,89 @@
 import React, { useCallback, useEffect, useRef, useState } from 'react'
 import { useSelector } from 'react-redux'
-import { Autocomplete, Box, InputAdornment, TextField, Typography } from '@mui/material'
+import { Autocomplete, InputAdornment, Stack, TextField, Typography } from '@mui/material'
 import { Close as CloseIcon, Search as SearchIcon } from '@mui/icons-material'
 import { searchListSelector } from '../../store/modules/search'
+import debounce from 'lodash/debounce'
 
 function SearchInput({ placeholder, onChange = () => null, onSelect = () => null, onHighlightChange }) {
   const [inputValue, setInputValue] = useState('')
   const [selected, setSelected] = useState(null)
-  const changeTimeoutRef = useRef(0)
+  const debouncedFuncRef = useRef(null)
   const options = useSelector(searchListSelector)
 
   const handleChange = e => {
-    clearTimeout(changeTimeoutRef.current)
     setInputValue(e?.target?.value)
-
-    changeTimeoutRef.current = setTimeout(() => {
-      onChange(e)
-    }, 500)
   }
 
-  const handleInputKeyDown = useCallback(
-    e => {
-      if ([38, 40].includes(e?.keyCode)) {
-        const optionNodesList = document.querySelectorAll('.MuiAutocomplete-popper [data-feature-id]')
-        const prevFocusedNode = document.querySelector('.MuiAutocomplete-popper .Mui-focused[data-feature-id]')
-        const prevFocusedIndex = +prevFocusedNode?.dataset?.optionIndex
-
-        let nextIndex = prevFocusedIndex + (e?.keyCode === 38 ? -1 : 1)
-        if (nextIndex === -1) {
-          nextIndex = optionNodesList.length - 1
-        } else if (nextIndex === optionNodesList.length) {
-          nextIndex = 0
-        }
-
-        const nextOptionNode = [...optionNodesList].find(node => +node?.dataset?.optionIndex === nextIndex)
-        const nextOption = options?.find(item => item?.id === nextOptionNode?.dataset?.featureId)
-
-        onHighlightChange(nextOption)
-      }
+  const handleHighlightChange = useCallback(
+    (event, option) => {
+      onHighlightChange(option)
     },
-    [options, onHighlightChange],
+    [onHighlightChange],
   )
 
-  const handleOptionMouseEnter = useCallback(
+  useEffect(() => {
+    onSelect(selected)
+  }, [onSelect, selected])
+
+  const handleKeyPress = useCallback(
     e => {
-      const featureId = e?.target?.dataset?.featureId
-      const option = options?.find(item => item?.id === featureId)
-      onHighlightChange(option)
+      if (e?.keyCode === 13) {
+        if (debouncedFuncRef.current?.cancel) {
+          debouncedFuncRef.current.cancel()
+        }
+
+        onChange(e?.target?.value)
+      }
     },
-    [options, onHighlightChange],
+    [onChange],
   )
```

Comment:
Now that this is further along, I'm getting a clearer idea of what behavior I'd like to see from the search field. I think it was the wrong idea to overload "Enter" to mean both "search now" and "select the highlighted option".

How does this sound for behavior?

- Don't do any special handling of key presses. Rely on `<Autocomplete>`'s default "Enter" behavior.

- When the input is focused
    - If there are options, show them.
    - If the input is blank, keep the options popup closed and clear the options list.
    - If the input is not blank and there are no results after its search query returns, show "No results".
    - I think these should be doable with a combination of `noOptionsText`, `onOpen`, `open`, and `openOnFocus`.

- When the input text changes
    - The autocomplete should show an indeterminate progress indicator on its right like https://mui.com/components/autocomplete/#load-on-open while it waits for results.
    - Unlike that example, it shouldn't replace the options with "Loading…" text. Previous options should remain while waiting for new options to load.

- When a user selects an option
    - The input text should remain as it is. It should not be cleared or set to the name of the selected option. This should make it easier for a user to revisit their last search after making a selection.

Reply:
I've added checkboxes to my previous comment to mark what parts aren't working right yet.

- The options popup shouldn't open or show "No results" if the input is empty.

- After applying the previous rule, the options popup should open whenever it's focused. It does if you click on the input but not if you tab to it.

- "No results" should only appear after a search has completed and returned no results.